imap_data_access/io.py

```diff
@@ -387,20 +387,30 @@ def reprocess(
         logger.debug("Received JSON: %s", items)
 
 
-def upload(file_path: Union[Path, str]) -> None:
+def upload(file_path: Union[Path, str], manually_reprocessed=False) -> None:
     """Upload a file to the data archive.
 
     Parameters
     ----------
     file_path : pathlib.Path or str
         Path to the file to upload.
+    manually_reprocessed : bool, optional
+        If True, the file is uploaded to s3 with the 'manually_reprocessed' tag.
+        Default is False.
     """
     file_path = Path(file_path).resolve()
     if not file_path.exists():
         raise FileNotFoundError(file_path)
 
+    # If the flag manually_reprocessed is set, we will add a tag to the upload
+    tags = (
+        f"?manually_reprocessed={str(manually_reprocessed).lower()}"
+        if manually_reprocessed
+        else ""
+    )
+
     # The upload name needs to be given as a path parameter
-    url = f"{imap_data_access.config['DATA_ACCESS_URL']}/upload/{file_path.name}"
+    url = f"{imap_data_access.config['DATA_ACCESS_URL']}/upload/{file_path.name}{tags}"
```

Comment:
Do we need to add this as a query param? I'm wondering if you can just inject it into the headers when PUTing the object and not bother with the queryParams at all? Or add it in the headers of your initial request in addition to the second and always be getting this information from the headers field.